healthtools/scrapers/base_scraper.py

```diff
@@ -0,0 +1,189 @@
+from bs4 import BeautifulSoup
+from cStringIO import StringIO
+from datetime import datetime
+import requests
+import re
+import json
+import hashlib
+
+
+class Scraper(object):
+    def __init__(self):
+        self.num_pages_to_scrape = None
+        self.site_url = None
+        self.fields = None
+        self.cloudsearch = None
+        self.s3 = None
+        self.s3_key = None
+        self.document_id = 0  # id for each entry, to be incremented
+        self.delete_file = None  # contains docs to be deleted after scrape
+        self.s3_historical_record_key = None  # s3 historical_record key
+
+    def scrape_site(self):
+        '''
+        Scrape the whole site
+        '''
+        self.get_total_number_of_pages()
+        all_results = []
+        delete_batch = []
+        skipped_pages = 0
+
+        print "Running {} ".format(type(self).__name__)
+        for page_num in range(1, self.num_pages_to_scrape + 1):
+            url = self.site_url.format(page_num)
+            try:
+                print "Scraping page %s" % str(page_num)
+                scraped_page = self.scrape_page(url)
+                entries = scraped_page[0]
+                delete_docs = scraped_page[1]
+
+                all_results.extend(entries)
+                delete_batch.extend(delete_docs)
+                print "Scraped {} entries from page {} | {}".format(len(entries), page_num, type(self).__name__)
+            except Exception as err:
+                skipped_pages += 1
+                print "ERROR: scrape_site() - source: {} - page: {} - {}".format(url, page_num, err)
+                continue
```

Comment:
If a page scrape doesn't work, we should retry 5 times and if not possible, quit the scrape for the specific site.

In this case if any page in doctors or foreign docs fails more than 5 times, the scrape should move to trying to scrape COs.

Counter of failure also resets to zero when there is successful retry.